.github/workflows/ci.yml

```diff
@@ -539,14 +539,15 @@ jobs:
           - name: macos-14
             runs-on: macos-14
             # No container, just install what we need directly.
-          - name: arm64
-            runs-on:
-              - self-hosted # must come first
-              - runner-group=${{ needs.setup.outputs.runner-group }}
-              - environment=${{ needs.setup.outputs.runner-env }}
-              - arm64
-              - os-family=Linux
-            container: gcr.io/iree-oss/base-arm64@sha256:9daa1cdbbf12da8527319ece76a64d06219e04ecb99a4cff6e6364235ddf6c59
+          # Disabled while we are short on arm64 runners. Could also make this opt-in, but the matrix complicates that.
```

Comment:
> I also explored options for making it opt-in, but support for conditional jobs in matrices within GitHub Actions is a complete mess.

See https://github.com/ScottTodd/iree/pull/15

Reusable workflow names are verbose (while also including _less_ information, somehow) and complicate the UI:

![image](https://github.com/user-attachments/assets/a50a8c19-c4bc-4b41-81ed-fb2a9a5435a3)
![image](https://github.com/user-attachments/assets/0d595d5d-dfa4-4d9c-8e0e-cd3a2c775af6)

Overriding the workflow names results in broken names for skipped jobs (seeing the job names is kind of the point...):

![image](https://github.com/user-attachments/assets/4df05473-15d1-43db-835e-cbb5ea1d137b)

That's not even getting into how we'd have `configure_ci.py` enumerate the jobs and decide on a filtering mechanism with git trailers.

Reply:
Ooh, maybe I can drop the matrix there, since the reusable workflow does basically the same thing.

Reply:
That works but is still too complicated in the UI: 
![image](https://github.com/user-attachments/assets/ecda1dde-21f1-4edd-be69-ca7d58d61426)

alternative is I guess to just fully undo the matrix changes to build_test_runtime: https://github.com/iree-org/iree/pull/17509 (or at least move the arm64 runtime build out of it)